packages/expo-web-browser/ios/EXWebBrowser/EXWebBrowser.m

```diff
@@ -105,13 +105,14 @@ - (dispatch_queue_t)methodQueue
     safariVC.preferredControlTintColor = [EXWebBrowser convertHexColorString:arguments[WebBrowserControlsColorKey]];
   }
   safariVC.delegate = self;
+  // By setting the modal presentation style to OverFullScreen, we disable the "Swipe to dismiss"
+  // gesture that is causing a bug where sometimes `safariViewControllerDidFinish` is not called.
+  // There are bugs filed already about it on OpenRadar.
+  [safariVC setModalPresentationStyle: UIModalPresentationOverFullScreen];
 
   // This is a hack to present the SafariViewController modally
```

Comment:
not related to this particular PR but we should include more context in comments like this, and err on the side of providing too much information. i don't personally understand the history behind this, if anyone does know please augment the comment :)